starboard/android/shared/audio_decoder.cc

```diff
@@ -108,15 +109,20 @@ void AudioDecoder::Decode(const InputBuffers& input_buffers,
 
   audio_frame_discarder_.OnInputBuffers(input_buffers);
 
+#if STARBOARD_ANDROID_SHARED_AUDIO_DECODER_VERBOSE
   for (const auto& input_buffer : input_buffers) {
     VERBOSE_MEDIA_LOG() << "T1: timestamp " << input_buffer->timestamp();
   }
+#endif
 
-  media_decoder_->WriteInputBuffers(input_buffers);
+  if (media_decoder_) {
```

Comment:
I don't know at a high enough level which edge case this could occur in, but if it does occur, it's possible that it's still necessary to call or remember the `consumed_cb` below to avoid hangups of other async logic. I don't have enough knowledge of the logic here to judge that it won't be necessary, only that dereferencing a null here would be bad, and there are crashes that appear they could be the result of calling the `WriteInputBuffers` method with a null object.